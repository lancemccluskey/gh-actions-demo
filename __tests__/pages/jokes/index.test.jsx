import { render, screen, fireEvent } from '@testing-library/react';
import Jokes from '../../../pages/jokes';

const testData = { data: { joke: 'testJoke', } };

describe('<Jokes />', () => {
  it('renders', () => {
    const { container } = render(<Jokes data={testData} />);

    expect(container).toMatchSnapshot();
  });

  it('should display Not implemented if Like clicked', () => {
    render(<Jokes data={testData} />);

    expect(screen.queryByText('Not implemented')).toBeNull();
    const likeButton = screen.getAllByRole('button')[0];
    fireEvent.click(likeButton);

    expect(screen.getByText('Not implemented')).toBeInTheDocument();
  });

  it('should display new joke if next clicked', async () => {
    fetch.resetMocks();

    render(<Jokes data={testData} />);

    fetch.mockResponseOnce(JSON.stringify({ joke: 'newjoke' }));

    const nextButton = screen.getAllByRole('button')[1];
    fireEvent.click(nextButton);

    await screen.findByText('newjoke');
  })
});
