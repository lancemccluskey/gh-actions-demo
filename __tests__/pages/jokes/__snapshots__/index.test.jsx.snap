// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Jokes /> renders 1`] = `
<div>
  <div
    class="css-1jnc401-Jokes"
  >
    <div
      class="css-1nuvgh6-Card"
    >
      <div
        class="css-13uxruf-Button"
      >
        <button
          class="css-uw5q0g-Button"
        >
          Like
        </button>
        <button
          class="css-uw5q0g-Button"
        >
          Next
        </button>
      </div>
    </div>
  </div>
</div>
`;
